docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -1,14 +1,574 @@
+
+### Create the SSO logic
+
+1. In your code editor, open the file `public\javascripts\ssoAuthES6.js`. It already has code that ensures that Promises are supported, even in Internet Explorer 11, and a call of Office.onReady to assign a handler to the add-in's only button.
+
+	> [!NOTE]
+    > As the name suggests, the ssoAuthES6.js uses JavaScript ES6 syntax because using `async` and `await` best shows the essential simplicity of the SSO API. When the localhost server is started, this file is transpiled to ES5 syntax so that the sample will run in Internet Explorer 11. 
+
+1. Add the following code below the Office.onReady method:
+
+    ```javascript
+    async function getGraphData() {
+        try {
+            
+            // TODO 1: Tell Office to get a bootstrap token from Azure AD.
+            
+            // TODO 2: Attempt to exhange the bootstrap token for an 
+            //         access token to Microsoft Graph.
+
+            // TODO 3: Handle case where Microsoft Graph requires an 
+            //         additional form of authentication.
+
+            // TODO 4: Use the access token in a call to Microsoft Graph 
+            //         or handle any error from the attempted token exchange.
+
+        }
+        catch(exception) {
+
+            // TODO 5: Respond to exceptions thrown by the call of 
+            //         OfficeRuntime.auth.getAccessToken.
+
+        }
+    }
+    ```
+
+1. Replace `TODO 1` with the following code. About this code, note:
+
+    - `OfficeRuntime.auth.getAccessToken` instructs Office to get a bootstrap token from Azure AD. A bootstrap token is similar to an ID token, but it has a `scp` (scope) property with the value `access-as-user`. This kind of token can be exchanged by a web application for an access token to Microsoft Graph.
+    - Setting the `allowSignInPrompt`option to true means that if no user is currently signed in to Office, then Office will open a popup signin prompt.
+    - Setting the `forMSGraphAccess` option to true signals to Office that the add-in intends to use the bootstrap token to get an access token to Micrsoft Graph, instead of just using it as an ID token. If the tenant administrator has not granted consent to the add-in's access to Microsoft Graph, then `OfficeRuntime.auth.getAccessToken` returns error **13012**. The add-in can respond by falling back to an alternative system of authorization, which is necessary because Office can prompt only for consent to the user's Azure AD profile, not to any Microsoft Graph scopes. The fallback authorization system requires the user to sign-in again and the user *can* be prompted to consent to Micrsoft Graph scopes. So, the `forMSGraphAccess` option ensures that the add-in won't make a token exchange that will fail due to lack of consent. (Since you granted administrator consent in an earlier step, this scenario won't happen for this add-in. But the option is included here anyway to illustrate a best practice.)
```

Comment:
```suggestion
    - Setting the `forMSGraphAccess` option to true signals to Office that the add-in intends to use the bootstrap token to get an access token to Micrsoft Graph, instead of just using it as an ID token. If the tenant administrator has not granted consent to the add-in's access to Microsoft Graph, then `OfficeRuntime.auth.getAccessToken` returns error **13012**. The add-in can respond by falling back to an alternative system of authorization, which is necessary because Office can prompt only for consent to the user's Azure AD profile, not to any Microsoft Graph scopes. The fallback authorization system requires the user to sign in again and the user *can* be prompted to consent to Micrsoft Graph scopes. So, the `forMSGraphAccess` option ensures that the add-in won't make a token exchange that will fail due to lack of consent. (Since you granted administrator consent in an earlier step, this scenario won't happen for this add-in. But the option is included here anyway to illustrate a best practice.)
```